server/interview/views.py

```diff
@@ -191,8 +195,111 @@ class NotifyInterview(APIView):
     def post(self, request):
         # TODO: Implement the notification service
         return Response({"detail": "Notification sent."})
+    
+class InterviewAll(APIView):
+    permission_classes = [IsAuthenticated]
+    # TODO: admin permission
+    def get(self, request):
+        interviews = Interview.objects.all()
+
+        # Check if there are no interviews
+        if not interviews:
+            return Response(
+                {"detail": "No interviews found."},
+                status=status.HTTP_404_NOT_FOUND
+            )
+        
+        # Prepare the response data
+        interview_data = []
+        for interview in interviews:
+            behavioral_questions = interview.behavioral_questions.all() if interview.behavioral_questions else []
+            for question in behavioral_questions:
+                print(question.prompt)
+            interview_data.append({
+                "interview_id": interview.interview_id,
+                "interviewer": interview.interviewer.username if interview.interviewer else "N/A",
```

Comment:
I would probably return None here instead of a string "N/A". Ditto for all of these.

Reply:
is interviewer/interviewee Optional? feels like it should never be `not interview.interviewer`

Reply:
change to use serializer